pkg/kubernetes/scripts/projects.js

```diff
@@ -298,6 +300,108 @@
         }
     ])
 
+    .factory('roleActions', [
+        '$modal',
+        function($modal) {
+            function addMember(namespace) {
+                return $modal.open({
+                    controller: 'MemberNewCtrl',
+                    templateUrl: 'views/add-member-role-dialog.html',
+                    resolve: {
+                        fields : function(){
+                            var fields = {};
+                            fields.namespace = namespace;
+                            return fields;
+                        }
+                    },
+                });
+            }
+            return {
+                addMember: addMember,
+            };
+        }
+    ])
+ 
+    .controller('MemberNewCtrl', [
+        '$q',
+        '$scope',
+        'projectData',
+        'projectPolicy',
+        'kubeLoader',
+        'kubeSelect',
+        'fields',
+        function($q, $scope, projectData, projectPolicy, loader, kselect, fields) {
+            var registryRoles = [{ ocRole: "registry-admin", displayRole :"Admin"},
+                { ocRole:"registry-editor", displayRole :"Push" },
+                { ocRole:"registry-viewer", displayRole :"Pull" }];
+
+            $scope.select = {
+                member: 'Select Members',
+                members: getAllMembers(),
+                displayRole: 'Select Role',
+                roles: registryRoles,
+                kind: "",
+                ocRole: "",
+            };
+            
+
+            var namespace = fields.namespace;
+            loader.watch("users");
```

Comment:
Not sure what the point of watching users and projects here is. The list of members on the scope won't be able to use anything new that it loads. If the expectation is that the caller will ensure the users and members are already loaded then these can just be removed. Otherwise the way you build the list of members needs to be reworked